### PR TITLE
Fix writing compressed CSVs in python3

### LINKED ISSUE #187: Py3: Encode strings when compressing
Another drive by report. Maybe I'll contribute someday :) This one should be easy

```python
In [1]: import odo

In [2]: with open('tst.csv', 'w') as f:
    f.write('a,1\nb,2\nc,3')
   ...:

In [3]: !cat tst.csv
a,1
b,2
c,3
In [4]: odo.odo('tst.csv', 'tst.csv.gz')
```

raises

```python
...
/usr/local/Cellar/python3/3.4.3/Frameworks/Python.framework/Versions/3.4/lib/python3.4/gzip.py in write(self, data)
    341         if len(data) > 0:
    342             self.size = self.size + len(data)
--> 343             self.crc = zlib.crc32(data, self.crc) & 0xffffffff
    344             self.fileobj.write( self.compress.compress(data) )
    345             self.offset += len(data)

TypeError: 'str' does not support the buffer interface
```

I believe that

```python
ipdb> zlib.crc32(data.encode('utf-8'), self.crc) & 0xffffffff
2324740828
```

Or something like `sys.defaultencoding` instead of utf-8 is the fix.

```python
In [5]: odo.__version__
Out[5]: '0.3.2'
```

### PR DESCRIPTION
**Pull request by cpcloud**
closes #187

### CODE REVIEW

**Comment by cpcloud** [comment]
@TomAugspurger look ok to you?

File: odo/backends/csv.py [hunk 1]
```diff
@@ -91,7 +92,10 @@ def append_dataframe_to_csv(c, df, dshape=None, **kwargs):
     encoding=kwargs.get('encoding', c.encoding)
 
     if ext(c.path) in compressed_open:
-        f = compressed_open[ext(c.path)](c.path, mode='a')
+        kwargs = dict(mode='at')
+        if sys.version_info[0] >= 3:
```
Comment (TomAugspurger) [inline odo/backends/csv.py]: `sys.version_info` is a named tuple so you can do `sys.version_info.major`

File: odo/backends/tests/test_csv.py [hunk 2]
```diff
@@ -314,3 +314,12 @@ def test_infer_header():
...
     assert CSV('foo.csv', sep=';').dialect['delimiter'] == ';'
+
+
+def test_csv_to_compressed_csv():
+    with tmpfile('.csv') as fn:
+        with open(fn, 'w') as f:
+            f.write('a,1\nb,2\nc,3')
+        with tmpfile('.csv.gz') as gfn:
+            result = odo(fn, gfn)
+            assert odo(result, list) == odo(fn, list)
```
Comment (TomAugspurger) [inline odo/backends/tests/test_csv.py]: Do these need to be `seek`ed back to the start? I don't know how odo handles opened files. Just worried you may be comparing 2 empty lists.

File: odo/backends/csv.py (hunk 1 above)
Comment (cpcloud) [inline odo/backends/csv.py]: alas, we have to support python 2.6, which doesn't implement `sys.version_info` as a `namedtuple`:

```python
Python 2.6.9 |Continuum Analytics, Inc.| (unknown, Jan 10 2014, 13:33:57)
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version_info
(2, 6, 9, 'final', 0)
```

File: odo/backends/tests/test_csv.py (hunk 2 above)
Comment (cpcloud) [inline odo/backends/tests/test_csv.py]: We're going straight from the filename like your example, so there's nothing to rewind.

Comment (cpcloud) [inline odo/backends/tests/test_csv.py]: In other words, we don't open the file handle until we're ready to convert. After we convert, the file handle is closed. We can call convert over and over without worrying about having a ton of files open.

Comment (TomAugspurger) [inline odo/backends/tests/test_csv.py]: Gotcha. I read `tmpfile` as a `TempraryFile` from the standard lib. Looks good!

**Comment by cpcloud** [comment]
cool, going to give a go on windows and will merge if that's successful

**Comment by cpcloud** [comment]
looks good on windows, merging

**Pull request merged**